src/foraging_gui/Foraging.py

```diff
@@ -2080,12 +2094,16 @@ def _set_metadata_enabled(self, enable: bool):
         self.Experimenter.setEnabled(enable)
         self.Tower.setEnabled(enable)
 
+    def _CheckBonsaiConnection(self):
```

Comment:
This doesn't check the connection, it only establishes a connection if `InitializeBonsaiSuccessfully` = 0. Shouldn't it always check?

Reply:
Yes. How about we merge this first? And I will create a separate pull request to solve this issue. This problem only occurs when the Bonsai and GUI are connected and then disconnected.

Reply:
ok

Reply:
https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/111